common/src/main/java/com/wynntils/models/lootruns/LootrunCompiler.java

```diff
@@ -201,6 +209,80 @@ private static Long2ObjectMap<List<ColoredPath>> generatePointsByChunk(LootrunPa
         return sampleByChunk;
     }
 
+    private static List<Vector3f> generateSimplifiedPoints(LootrunPath raw, double tolerance) {
+        List<Vector2d> points = new ArrayList<>();
+
+        // y is discarded in the process, as map doesn't show height info
+        for (Vec3 point : raw.points()) {
+            points.add(new Vector2d(point.x, point.z));
+        }
+
+        List<Vector2d> intermediate = simplify(points, tolerance);
+        List<Vector3f> result = new ArrayList<>();
+
+        for (Vector2d vec : intermediate) {
+            result.add(new Vector3f((float) vec.x, (float) 0, (float) vec.y));
+        }
+
+        return result;
```

Comment:
Is this remapping necessary? Why not just store simplified points as Vector2f?

Reply:
Why was this marked as resolved? I have the same question.

Reply:
>I know that the code could be simplified a little bit, but I think I (or anyone else) will go ahead and do that later when this is also rendered on the mini-map. While it is only on the main map, the bit higher allocation is acceptable.

the reason it was marked as resolved, but I could do it myself I suppose